crates/api/src/builder/api.rs

```diff
@@ -1646,7 +1646,7 @@ fn sanity_check_block_submission(
     payload_attributes: &PayloadAttributesUpdate,
     chain_info: &ChainInfo,
 ) -> Result<(), BuilderApiError> {
-    let expected_timestamp = chain_info.genesis_time_in_secs + (bid_trace.slot * chain_info.seconds_per_slot);
+    let expected_timestamp = chain_info.genesis_time_in_secs + (bid_trace.slot * chain_info.seconds_per_slot) + chain_info.context.genesis_delay;
```

Comment:
I think we can use this instead: 
https://github.com/lu-bann/taiyi/blob/9382993fc17b1256c067ddbfd861ce9a23dd7673/crates/preconfer/src/preconf_api/mod.rs#L63

Reply:
Yeah it would fail for all testnets. 
`genesis_time` = `min_geneisis_time` + `genesis_delay`
this holds true only for testnets, not mainnet

Reply:
Make perfect sense!!